QueueArray.java

```diff
@@ -0,0 +1,194 @@
+
+/**
+ *
+ * @author cuong.hq1
+ */
+public class QueueArray {
+
+    private int capacity;
+    private int[] queue;
+
+    // Indicate front indicator of queue, alway point at non-value index,
+    // next index of front indicator will be the front value of queue.
+    private int front;
+
+    // Indicate rear (tail) of queue.
+    private int rear;
+
+    /**
+     * Initialize queue with the current capacity of the queue to be k. **
+     * Instead of initialize queue with capacity of k, add a extra element in
+     * queue array to prevent confuse situation when rear indicator and front
+     * one at a same position, this will bring.
+     *
+     */
+    public void init(int capacity) {
+        this.capacity = capacity + 1;
+        queue = new int[capacity];
+        front = 0;
+        rear = 0;
+    }
+
+    public QueueArray(int capacity) {
+        this.init(capacity);
+    }
```

Comment:
`init` should be a private method. But it should better not exist, since its role is the same as the constructor. Putting the initialization code in the constructor is better.